cmd/erasure-multipart.go

```diff
@@ -199,38 +199,57 @@ func (er erasureObjects) cleanupStaleUploadsOnDisk(ctx context.Context, disk Sto
 	readDirFn(pathJoin(drivePath, minioMetaMultipartBucket), func(shaDir string, typ os.FileMode) error {
 		readDirFn(pathJoin(drivePath, minioMetaMultipartBucket, shaDir), func(uploadIDDir string, typ os.FileMode) error {
 			uploadIDPath := pathJoin(shaDir, uploadIDDir)
-			fi, err := disk.ReadVersion(ctx, "", minioMetaMultipartBucket, uploadIDPath, "", ReadOptions{})
-			if err != nil {
+			var modTime time.Time
+			// Upload IDs are of the form base64_url(<UUID>x<UnixNano>), we can extract the time from the UUID.
+			if b64, err := base64.RawURLEncoding.DecodeString(uploadIDDir); err == nil {
+				if split := strings.Split(string(b64), "x"); len(split) == 2 {
```

Comment:
will 'x' never confused the base64 decoder ? because x can exist in a base64 output

Reply:
@klauspost we need to perhaps add the right "base64" key length to be split instead of `x<UnixNano>` in this case key length for base64 must be constant

Reply:
@vadmeste `x` cannot be present in a UUID. The split is on the *decoded* upload id, not the encoded data.

@harshavardhana Why? It is just an ID.